server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -1155,6 +1134,35 @@ protected Node(
                 taskHeaders,
                 tracer
             );
+            final QueryGroupService queryGroupService = new QueryGroupService(transportService.getLocalNode(), clusterService); // We will
+                                                                                                                                // need to
+                                                                                                                                // replace
+                                                                                                                                // this with
+                                                                                                                                // actual
+            // instance of the
+            // queryGroupService
+            queryGroupServiceSetOnce.set(queryGroupService);
+
+            final QueryGroupRequestOperationListener queryGroupRequestOperationListener = new QueryGroupRequestOperationListener(
+                queryGroupService,
+                threadPool
+            );
```

Comment:
I am assuming these changes are needed for collecting the stats? Ideally, the changes for collecting stats and the API should be in separate PRs. Also, @kaushalmahi12 is adding QueryGroupService as part of his PR. I am assuming one of you will need to resolve conflict depending on which PR is merged first.